### PR TITLE
Add workflow for e2e tests

### PR DESCRIPTION
**Pull request by bogdan-luzmo**

### CODE REVIEW

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -1,33 +1,43 @@
 name: E2E tests
 on:
+  workflow_call:
+  workflow_dispatch:
   push:
-    branches: [ main ]
+    branches: [ main, feat/workflow-for-e2e ]
```
Comment (KrijnL) [inline .github/workflows/e2e.yml]: **Question:** as I understand it, we only want to run these tests after a deploy? This already happens on push to the main branch, so probably we can remove the `push` directive here.

File: .github/workflows/e2e.yml [hunk 2]
```diff
@@ -1,33 +1,43 @@
 name: E2E tests
 on:
+  workflow_call:
+  workflow_dispatch:
```
Comment (KrijnL) [inline .github/workflows/e2e.yml:4]: **Nitpick:** I think we only need `workflow_call` to be able to call this workflow from the deployment one. `workflow_dispatch` is only necessary if we want to trigger it manually.

**Thought:** We can also run it on a schedule (e.g. daily) by using the [schedule directive](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#schedule). Perhaps in that case we should send the results to Slack somehow? Or report them somewhere else that we're aware?

Comment (bogdan-luzmo) [reply .github/workflows/e2e.yml:4]: Schedule and alert on slack is good idea.
I've added workflow_dispatch just in case. But if you want I can remove it

Comment (KrijnL) [reply .github/workflows/e2e.yml:4]: No, it's fine that it's there. Perhaps we want to run the tests manually at some point. :slightly_smiling_face:

**Pull request merged**